util/statvar_dcid_generator.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import re
+
+
+class StatVarDcidGenerator:
+    """Helper class to generate dcid for a statistical variable."""
+    STAT_TYPE_LOOKUP = {'meanValue': 'mean', 'medianValue': 'median'}
+
+    def __init__(self):
+        pass
+
+    def capitalize_word(self, word):
+        """Changes the first letter of a word to upper case."""
+        if (len(word) != 0):
+            return (word[0].upper() + word[1:])
+        else:
+            return None
+
+    def append_to_dcid(self, dcid, value, end_underscore=False):
+        """Append a value to a dcid."""
+        dcid += self.capitalize_word(value)
+        if (end_underscore == True):
+            dcid += '_'
+
+        return dcid
+
+    def generate_quantity_range_name(self, s):
+        """Generate a name for a quantity range.
+        Expected syntax of s - [<lower_limit> <upper_limit> <quantity>]
+        """
+        str_list = s.split()
+
+        lower_limit = str_list[0][1:]
+        upper_limit = str_list[1]
+        quantity = str_list[2][:-1]
+
+        quantity = self.capitalize_word(quantity)
+        if (upper_limit == '-'):
+            return f"{lower_limit}OrMore{quantity}"
+        elif (lower_limit == '-'):
+            return f"{upper_limit}OrLess{quantity}"
+        else:
+            return f"{lower_limit}to{upper_limit}{quantity}"
+
+    def generate_quantity_name(self, s):
+        """Generate a name for a quantity.
+        Expected syntax of s- [<value> <quantity>]
+        """
+        str_list = s.split()
+
+        value = str_list[0][1:]
+        quantity = str_list[1][:-1]
+
+        quantity = self.capitalize_word(quantity)
+        return f"{value}{quantity}"
+
+    def get_stat_var_name(self, stat_var_dict, ignore_props=None):
+        """Generates the dcid given a statistical variable
+        
+        Arguments:
+        stat_var_dict -- A dictionary with property:values of the statistical variable as key-value pairs
+        ignore_props -- A list of properties to ignore from stat_var_dict when generating the dcid
```

Comment:
What's an example of these ignore_props?

Reply:
The ignore_props can be used to account for dependent PVs which we want to ignore when generating the dcid.

For example in the following statVar,
```
populationType: Person
measuredProperty: count
statType: measuredValue
healthInsurance: NoHealthInsurance
armedForceStatus: Civilian
institutionalization: USC_NonInstitutionalized
```
Since the ```healthInsurance``` property indicates they are ```Civilian``` and ```USC_NonInstitutionalized```, ignore_props can be ```['armedForceStatus', 'institutionalization']```. During the dcid generation process, these PVs will not be considered.